Templates/AppSource App/.github/workflows/CICD.yaml

```diff
@@ -206,8 +227,8 @@ jobs:
         uses: actions/deploy-pages@v4
 
   Deploy:
-    needs: [ Initialization, Build ]
-    if: always() && needs.Build.result == 'Success' && needs.Initialization.outputs.environmentCount > 0
+    needs: [ Initialization, Build, BuildPP ]
+    if: (!cancelled()) && needs.Build.result != 'Failure' && needs.BuildPP.result != 'Failure' && needs.Initialization.outputs.environmentCount > 0
```

Comment:
I think the check for 'Success' is mandatory. Otherwise, the check has to include all build jobs.

Or wouldn't the `Deploy` job run if the `Build` job was skipped (because, say, `Build1` failed)?

Reply:
Will have to test this in a multiproject repo

Reply:
Right you are - 
![image](https://github.com/microsoft/AL-Go/assets/10775043/dd869cba-b176-4a76-b065-f0a70bef4460)

Will investigate further - good catch

Reply:
Fixed this (also in CheckForUpdates when building CI/CD) - needs to check all jobs now